Assets/MRTK/Core/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -300,12 +298,148 @@ internal static void EnsureMSBuildForUnity()
+                // Update the manifest's scoped registries.
+                manifest.scopedRegistries = scopedRegistries.ToArray();
+            }
+
+            // Attempt to find the msbuild for unity package
+            // This loop also identifies the lines that enclose the scoped registry collection.
+            int registriesStartIndex = -1;
+            int registriesEndIndex = -1;
+            int packageToRemove = -1;
+            for (int i = 0; i < manifestFileLines.Count; i++)
+            {
+                if (manifestFileLines[i].Contains("\"scopedRegistries\":"))
+                {
+                    registriesStartIndex = i;
+                }
+                if (manifestFileLines[i].Contains("],") && (registriesStartIndex != -1) && (registriesEndIndex == -1))
+                {
+                    registriesEndIndex = i;
+                }
+                if (manifestFileLines[i].Contains(MSBuildPackageName))
+                {
+                    packageToRemove = i;
+                    break;
+                }
+            }
+            if (packageToRemove != -1)
+            {
+                manifestFileLines.RemoveAt(packageToRemove);
+            }
+
+            // Update the manifest file
+            if ((registryToRemove == -1) &&
+                (packageToRemove == -1))
+            {
+                // No changes needed.
+                return;
+            }
+
+            WriteManifest(
+                manifest,
+                manifestFileLines,
+                registriesStartIndex,
+                registriesEndIndex);
+        }
+
+        private static PackageManifest LoadManifest(out List<string> manifestFileLines)
+        {
+            PackageManifest manifest = null;
+            manifestFileLines = new List<string>();
+
+            if (string.IsNullOrWhiteSpace(ManifestFilePath))
+            {
+                return null;
+            }
+
+            // Read the package manifest into a list of strings (for easy finding of entries)
+            // and then deserialize.
+            using (FileStream manifestStream = new FileStream(ManifestFilePath, FileMode.Open, FileAccess.Read))
+            {
+                using (StreamReader reader = new StreamReader(manifestStream))
+                {
+                    // Read the manifest file a line at a time.
+                    while (!reader.EndOfStream)
+                    {
+                        string line = reader.ReadLine();
+                        manifestFileLines.Add(line);
+                    }
+
+                    // Go back to the start of the file.
+                    manifestStream.Seek(0, 0);
+
+                    // Deserialize the scoped registries portion of the package manifest.
+                    manifest = JsonUtility.FromJson<PackageManifest>(reader.ReadToEnd());
+                }
+            }
+
+            return manifest;
+        }
+
+        private static void WriteManifest(
```

Comment:
Nit: Add doccomments for this (the most interesting thing here is specifying how the start/end index are used i.e. they will be omitted)

Reply:
sounds good